grammar/TypeQL.g4

```diff
@@ -113,13 +113,19 @@ variable_concept      :   VAR_  IS  VAR_  ;
 variable_type         :   type_any    type_constraint ( ',' type_constraint )*  ;
 type_constraint       :   ABSTRACT
                       |   SUB_        type_any
-                      |   OWNS        type         ( AS type )? ( IS_KEY )?
-                      |   RELATES     type         ( AS type )?
+                      |   OWNS        type         ( AS type )?     owns_annotations //value_annotations
+                      |   RELATES     type         ( AS type )?   //relates_annotations
                       |   PLAYS       type_scoped  ( AS type )?
                       |   VALUE       value_type
                       |   REGEX       STRING_
                       |   TYPE        label_any
+                      // value_annotations
                       ;
+//annotations           :   ( ANNOTATION_KEY )?   ( ANNOTATION_UNIQUE )?  ( ANNOTATION_RANGE | ANNOTATION_REGEX )?     ;
+
+owns_annotations      :   ( ANNOTATION_KEY )?   ( ANNOTATION_UNIQUE )?          ;
```

Comment:
@haikalpribadi do you think we should use a generic `annotations` block that can be any annotations, and then do the validation once we construct the TypeQL objects (see the commented `annotations` above, which we would apply using:
```as OWNS type (as type)? annotations```

In the future we will have long list of annotations (range, cardinality, regex, etc.) that are possible, but only some will be applicable to `owns`, some will be applicable to `plays` only, some will be applicable to the variable directly... given this I think we should have `owns_annotations` separate, then add over time.

Reply:
What if we can provide more than one annotation? 🤔 

Also, I think the it should be `annotation_own` and `annotation_relation`, no?

Reply:
Yes my bad re. annotation_owns, was prototyping too quickly! I think it would be on the constraint eg. `annotation_owns` and `annotation_relates` rather than `annotation_own` and `annotation_relation` though.

This mechanism does allow us to provide more than one annotation. The alternative is inline:
```
OWNS type ( AS type )? ( ANNOTATION_KEY )? ( ANNOTATION_UNIQUE )?
```
which also works but will overflow the line and isn't as nice to read.

Reply:
Why the overflow? You mean the line becomes too long horizontally?

Reply:
yes

Reply:
why not wrap it in another rule? like what you had before but better named?

Reply:
you mean the entire row for `type_owns_constraint`, or just keep the `OWNS type` as is and add a new `annotations_owns = ANNOTATION_KEY )? ( ANNOTATION_UNIQUE )?` similar to what I suggested?

Reply:
`(annotations_owns = ANNOTATION_KEY )? ( ANNOTATION_UNIQUE )?` enforces an order on the annotations
`@key @unique` parses, whereas `@unique @key` does not.